domains/Discussions/Policies/AnswerPolicy.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+
+namespace Domains\Discussions\Policies;
+
+
+use Domains\Accounts\Models\User;
+use Domains\Discussions\Models\Answer;
+use Domains\Discussions\Models\Question;
+use Illuminate\Auth\Access\HandlesAuthorization;
+
+class AnswerPolicy
+{
+    use HandlesAuthorization;
+
+    public function update(User $user, Answer $answer, Question $question): bool
+    {
+        return $answer->author->is($user) && $answer->question->is($question);
```

Comment:
With the scope we're applying in the controller, there's no need to check that the answer is related to the question. The generated query already checks that, implicitly.

Reply:
I was pretty confused that but I think I fixed what you wanted! Please look at my last commit.
At some point in the past I remember reading to do this according to that, but I adjusted this point anyway.

Reply:
Yes, it was an oversight on my part as I didn't quite notice the scope was already validating that the given answer must be associated with the given question.